include/alpaka/mem/buf/cpu/Copy.hpp

```diff
@@ -181,12 +181,13 @@ namespace alpaka
                             {
                                 meta::ndLoopIncIdx(
                                     extentWithoutInnermost,
-                                    [&](vec::Vec<DimMin1, ExtentSize> const & idx)
+                                    [&] ALPAKA_FN_HOST_ACC (vec::Vec<DimMin1, ExtentSize> const & idx)
                                     {
-                                        std::memcpy(
-                                            reinterpret_cast<void *>(this->m_dstMemNative + (vec::cast<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>())),
-                                            reinterpret_cast<void const *>(this->m_srcMemNative + (vec::cast<SrcSize>(idx) * srcPitchBytesWithoutOutmost).foldrAll(std::plus<SrcSize>())),
-                                            static_cast<std::size_t>(this->m_extentWidthBytes));
+                                        auto dst = reinterpret_cast<uint8_t *>(this->m_dstMemNative + (vec::cast<DstSize>(idx) * dstPitchBytesWithoutOutmost).foldrAll(std::plus<DstSize>()));
```

Comment:
I would think that we could declare this variable as `auto const` because the const should apply to the pointer an not to the value but I am not 100% sure.

Reply:
Indeed, `auto const` would make `dst` have type `uint8_t * const`, as opposed to `uint8_t` in the original code.